packages/types/src/feedback/index.ts

```diff
@@ -0,0 +1,79 @@
+import type { ComponentType } from 'preact';
+import type { Attachment } from '../attachment';
+import type { Integration } from '../integration';
+
+import type {
+  FeedbackCallbacks,
+  FeedbackGeneralConfiguration,
+  FeedbackTextConfiguration,
+  FeedbackThemeConfiguration,
+} from './config';
+
+export type { FeedbackFormData } from './form';
+
+import type { FeedbackEvent, SendFeedback, UserFeedback } from './sendFeedback';
+export type { FeedbackEvent, UserFeedback, SendFeedback };
+
+/**
+ * The integration's internal `options` member where every value should be set
+ */
+export interface FeedbackInternalOptions
+  extends FeedbackGeneralConfiguration,
+    FeedbackThemeConfiguration,
+    FeedbackTextConfiguration,
+    FeedbackCallbacks {}
+
+export interface FeedbackDialog {
+  /**
+   * The HTMLElement that is containing all the form content
+   */
+  el: HTMLElement;
+
+  /**
+   * Insert the Dialog into the Shadow DOM.
+   *
+   * The Dialog starts in the `closed` state where no inner HTML is rendered.
+   */
+  appendToDom: () => void;
+
+  /**
+   * Remove the dialog from the Shadow DOM
+   */
+  removeFromDom: () => void;
+
+  /**
+   * Open/Show the dialog & form inside it
+   */
+  open: () => void;
+
+  /**
+   * Close/Hide the dialog & form inside it
+   */
+  close: () => void;
+}
+
+export interface FeedbackModalIntegration extends Integration {
+  createDialog: (props: {
+    options: FeedbackInternalOptions;
+    screenshotIntegration: FeedbackScreenshotIntegration | undefined;
+    sendFeedback: SendFeedback;
+    shadow: ShadowRoot;
+  }) => FeedbackDialog;
+}
+
+export interface FeedbackScreenshotIntegration extends Integration {
+  createInput: (
+    h: any,
+    dialog: FeedbackDialog,
+  ) => {
+    /**
+     * The preact component
+     */
+    input: ComponentType<{ onError: (error: Error) => void }>;
```

Comment:
Can do.

I put the package under `devDependencies` to see if it passes with y'all, but i can play with it and find a middle-ground or i'd even be happy with `input: Function` or `input: unknown` at the end of the day to get things moving. 

Vendoring the types might be annoying because I think a large number of types would need to come in and i didn't see `@types/preact` out there to use.

Reply:
yeah ideally we can avoid the preact dep here because everything in sdk world relies on `@sentry/types` (react native and electron as well outside of this repo).